win32ss/user/user32/misc/exticon.c

```diff
@@ -323,6 +323,42 @@ static UINT ICO_ExtractIconExW(
 	}
 	CloseHandle(fmapping);
 
+#ifdef __REACTOS__
+    /* Check if we have a min size of 2 headers RIFF & 'icon'
+     * at 8 chars each plus an anih header of 36 byptes.
+     * Also, is this resource an animjated icon/cursor (RIFF) */
+    if ((fsizel >= 52) && !memcmp(peimage, "RIFF", 4))
+    {
+        UINT anihOffset;
+        /* Get size of the animation data */
+        ULONG uSize = MAKEWORD(peimage[4], peimage[5]);
+
+        /* Check if uSize is reasonable with respect to fsizel */
+        if (uSize > fsizel)
+            goto end;
+
+        /* Search for 'anih' indicating animation header */
+        for (anihOffset = 0; anihOffset < uSize; anihOffset++)
+        {
+            if (memcmp(&peimage[anihOffset], "anih", 4) == 0)
```

Comment:
If there's no `"anih"`?

Reply:
Maybe 2 more checks are needed.
1. The flag that indicate whether found or not.
2. When found, whether the trailing data is bounded or not.

Reply:
Check new file with RIFF followed my no anih in 200 bytes is now excluded.

Reply:
Excuse me, but... 

How about the case of 100 bytes and no anih?

Reply:
https://en.m.wikipedia.org/wiki/Interval_(mathematics)

Initially, the boundary range is `[0, fsizel)`.
The system checks anih in range `[0, uSize + 4)`.
So `uSize + 4 <= fsizel`.